### PR TITLE
Bug 1977981: Refactor OCS Dashboard to Support ODF Dashboard Abstraction

### PR DESCRIPTION
**Pull request by bipuladh**
- Automatically redirect to any tab based on components present.
- Remove usage of Extensions for Cards and Tabs.

### CODE REVIEW

**Comment by SanjalKatiyar** [comment]
LGTM.
Tested for following cases:
If it is an external mode OCS cluster --> First Tab was "Block and File" and both "Block and File" and "Object" tabs were present.
If it is an internal mode OCS cluster --> First Tab was "Block and File" and both "Block and File" and "Object" tabs were present.
If NooBaaSystem is there, but, StorageCluster is not --> Only "Object" was present and pre-selected.

Only issue is that it is talking some time to dashboard tabs to appear on re-load (which is already an known issue).

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  history,
+}) => {
+  const isIndependent = useFlag(OCS_INDEPENDENT_FLAG);
+  const isObjectServiceAvailable = useFlag(MCG_FLAG);
+  const isCephAvailable = useFlag(CEPH_FLAG);
+  const [pages, setPages] = React.useState<Page[]>([]);
+  const { t } = useTranslation();
+  const title = t('ceph-storage-plugin~OpenShift Container Storage Overview');
+
+  const location = useLocation();
```
Comment (vbnrh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx:137]: Add typing to it if possible

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+  const isOCS = location.pathname.includes('ocs-dashboards');
```
Comment (vbnrh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx:139]: i see `ocs-dashboards` being hardcoded in lots of places. 
we can extract it out in a common place and use it

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+  const showInternalDashboard = !isIndependent && isCephAvailable;
+
+  const internalPage = {
+    href: 'block-file',
+    name: t('ceph-storage-plugin~Block and File'),
+    component: () => <PersistentInternalDashboard />,
+  };
+  const externalPage = {
+    href: 'block-file',
```
Comment (vbnrh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx:149]: remove hardcoded references to common spot.

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  const title = t('ceph-storage-plugin~OpenShift Container Storage Overview');
+
+  const location = useLocation();
+
+  const isOCS = location.pathname.includes('ocs-dashboards');
+
+  const showInternalDashboard = !isIndependent && isCephAvailable;
+
+  const internalPage = {
+    href: 'block-file',
```
Comment (vbnrh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx:144]: remove hardcoded references to common spot.

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    name: t('ceph-storage-plugin~Block and File'),
+    component: () => <PersistentInternalDashboard />,
+  };
+  const externalPage = {
+    href: 'block-file',
+    name: t('ceph-storage-plugin~Block and File'),
+    component: () => <PersistentExternalDashboard />,
+  };
+  const objectPage = {
+    href: 'object',
```
Comment (vbnrh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx:154]: ditto

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+import DataConsumptionCard from './object-service/data-consumption-card/data-consumption-card';
+import { default as ObjectActivityCard } from './object-service/activity-card/activity-card';
+import { ResourceProvidersCard } from './object-service/resource-providers-card/resource-providers-card';
+import { OCS_INDEPENDENT_FLAG, MCG_FLAG, CEPH_FLAG } from '../../features';
+
+const convertToCard = (Card: React.ComponentType): GridDashboardCard => ({ Card });
+
+const isPagePresent = (pages: Page[], page: Page): boolean =>
+  pages.some((p) => page.href === p.href);
+
```
Comment (vbnrh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx:48]: Add a comment on top of this for how the sorting works. Would be useful for future reads

**Review by vbnrh — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx (hunk 1 above)
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/ocs-system-dashboard.tsx:137]: The authors of this library haven't provided types unfortunately.

**Pull request merged**